packages/web3-react/src/components/AlephiumConnect.tsx

```diff
@@ -121,20 +117,76 @@ function tryGetNodeProvider(): NodeProvider | undefined {
   }
 }
 
-export const AlephiumConnectProvider: React.FC<{ children?: React.ReactNode }> = ({ children }) => {
+export const AlephiumConnectProvider: React.FC<{
+  network: NetworkId
+  addressGroup?: number
+  keyType?: KeyType
+  children?: React.ReactNode
+}> = ({ network, addressGroup, keyType, children }) => {
   // Only allow for mounting AlephiumConnectProvider once, so we avoid weird global
   // state collisions.
   const context = useContext(AlephiumConnectContext)
   if (context) {
     throw new Error('Multiple, nested usages of AlephiumConnectProvider detected. Please use only one.')
   }
 
-  const [account, setAccount] = useState<Account & { network: NetworkId }>()
+  const lastConnectedAccount = useMemo(() => {
+    const result = getLastConnectedAccount()
+    if (result === undefined) {
+      return undefined
+    }
+    if (
+      result.network === network &&
+      (addressGroup === undefined || result.account.group === addressGroup) &&
+      (keyType === undefined || result.account.keyType === keyType)
+    ) {
+      return result
+    }
+  }, [network, addressGroup, keyType])
+
+  const [account, setAccount] = useState<Account | undefined>(lastConnectedAccount?.account)
   const [signerProvider, setSignerProvider] = useState<SignerProvider | undefined>()
 
+  const updateAccount = useMemo(() => {
+    return (newAccount: Account | undefined) => {
+      setAccount((prev) => (prev?.address === newAccount?.address ? prev : newAccount))
+    }
+  }, [setAccount])
+
+  useEffect(() => {
+    const func = async () => {
+      if (lastConnectedAccount === undefined) return
```

Comment:
Currently we auto connect based on `lastConnectedAccount.connectorId`, which means if `lastConnectedAccount` doesn't exist, we won't automatically reconnect even if we can. In wagmi it will retry the autoconnect for each connector if the last connector id does not exist. The code is here: https://github.com/wagmi-dev/wagmi/blob/4c614d524c7b0ac5b6817629468f5931815c1cd1/packages/core/src/config.ts#L243. Perhaps we can follow the same design, WDYT?

Reply:
Yes, we could do the same.